dandiapi/api/models/version.py

```diff
@@ -83,15 +83,46 @@ def make_version(cls, dandiset: Dandiset = None) -> str:
     def copy(cls, version):
         return Version(dandiset=version.dandiset, metadata=version.metadata)
 
+    @classmethod
+    def citation(cls, metadata):
+        year = datetime.datetime.now().year
+        name = metadata['name']
+        url = metadata['url']
+        # If we can't find any contributors, use this citation format
+        citation = f'{name} ({year}). Online: {url}'
+        if 'contributor' in metadata and metadata['contributor']:
+            cl = '; '.join(
+                [
+                    val['name']
+                    for val in metadata['contributor']
+                    if 'includeInCitation' in val and val['includeInCitation']
+                ]
+            )
+            if cl:
+                citation = f'{cl} ({year}) {name}. Online: {url}'
+        return citation
+
     def _populate_metadata(self):
```

Comment:
this is only called for `draft` right? 

for any published dandiset, we would return the stored metadata without any changes. i.e. on publish, the metadata record returned by this + any published added info would be stored as the dandiset metadata. i.e. the metadata becomes static.

Reply:
This method is only called when metadata is saved. As long as you don't somehow modify the metadata for the published version, the metadata won't be modified.